manifests/profile/www_lib/vhosts/press.pp

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# nebula::profile::www_lib::vhosts::press
+#
+# press virtual host
+#
+# @example
+#   include nebula::profile::www_lib::vhosts::press
+class nebula::profile::www_lib::vhosts::press (
+  String $prefix,
+  String $domain,
+  String $ssl_cn = 'www.press.umich.edu',
+  String $docroot = '/www/www.press/public'
+) {
+  $servername = "${prefix}www.press.umich.edu"
+
+  file { "${apache::params::logroot}/press":
+    ensure => 'directory'
+  }
+
+  nebula::apache::www_lib_vhost { 'press-http':
+    servername     => $servername,
+    docroot        => $docroot,
+    logging_prefix => 'press/',
+
+    rewrites       => [
+      {
+        rewrite_rule => '^(.*)$ https://%{HTTP_HOST}$1 [L,R]'
+      },
+    ],
+
+    directories    => [
+      {
+        provider      => 'directory',
+        path          => $docroot,
+        options       => 'FollowSymLinks',
+        allowoverride => 'None',
+        require       => $nebula::profile::www_lib::apache::default_access,
+      },
+    ],
+  }
+
+  nebula::apache::www_lib_vhost { 'press-https':
+    servername     => $servername,
+    docroot        => $docroot,
+    logging_prefix => 'press/',
+    ssl            => true,
+    ssl_cn         => $ssl_cn,
+    setenv         => ['HTTPS on'],
+
+    aliases        => [
+      { scriptalias => '/script/press', path => '/www/www.press/script/press' }
+    ],
+
+    rewrites       => [
+      {
+        rewrite_cond => [
+          '%{DOCUMENT_ROOT}/%{REQUEST_URI} !-f',
+          '%{REQUEST_URI} !^/cosign',
+          '%{REQUEST_URI} !^/script/press',
+        ],
+        rewrite_rule => '^(.*) /script/press$1 [L,PT]'
+      }
+    ],
+
+    directories    => [
+      {
+        provider      => 'directory',
+        path          => $docroot,
+        options       => 'FollowSymLinks',
+        allowoverride => 'None',
+        require       => $nebula::profile::www_lib::apache::default_access,
+      }, {
+        provider   => 'location',
+        path       => '/script/press',
+        options    => 'None',
+        require    => $nebula::profile::www_lib::apache::default_access,
+        sethandler => 'fcgid-script',
```

Comment:
This requires adding mod_fcgid to our apache config. The alternative is to switch to mod_proxy_fcgi as we did for imgsrv in HT. That is also how we are supporting PHP 7.3. @aelkiss Had you thought about this for the Press?

Reply:
mod_fcgid isn't available on stretch, I think, so I was expecting that we would use mod_proxy_fcgi.

Reply:
Mojolicious is Plack/PSGI-ready, so we set this up as a plackup service and very-much-like-our-ruby rewrite/proxy. Nothing FCGI-specific needed for this vhost.